src/cmd/ksh93/tests/read.exp

```diff
@@ -30,6 +30,18 @@ expect -ex "\r\ngimme input > "
 expect -ex "\r\ngot ''\r\n"
 expect_prompt
 
+#-p prompt Write prompt on each line before reading.
+send "read -p 'custom_prompt> ' foo\r"
+expect -ex "custom_prompt> "
+send [ctrl C]
+
+#-v When reading from a terminal the value of the first variable is displayed and used as a default
+# value.
+send "set foo bar\r"
+send "read -v foo\r"
+expect -ex "bar"
```

Comment:
Same as above. Here you're actually matching the "bar" from `set foo bar` which does not do what you intended. I'd also recommend using `read -p 'custom_prompt> ' -v foo` to make matching the prompt and default value clearer.